examples/indexing_routines.py

```diff
@@ -0,0 +1,335 @@
+        space = (5 * N * N + N) * np.dtype(int).itemsize / 1073741824
+        print("Total Size:       " + str(space) + " GB")
+        print("Elapsed Time for advanced_indexing: " + str(total) + " ms")
+        average = total / steps
+        print("Average            :     " + str(average) + " ms")
+        print(
+            "FLOPS/s:                 "
+            + str(flops / (average * 1e6))
+            + " GFLOPS/s"
+        )
+        print(
+            "bandwidth:               "
+            + str(space * 1000 / (average))
+            + " GB/s"
+        )
+    return total
+
+
+def run_indexing_routines(
+    N,
+    steps,
+    timing,
+    verbose,
+    routine,
+):
+    # simple operation to warm up the library
+    assert not math.isnan(np.sum(np.zeros((N, N)).dot(np.zeros((N,)))))
+    gc.collect()
+    time = 0
+    if routine == "diagonal" or routine == "all":
+        time += compute_diagonal(steps, N, timing)
+    if routine == "choose" or routine == "all":
+        time += compute_choose(steps, N, timing)
+    if routine == "repeat" or routine == "all":
+        time += compute_repeat(steps, N, timing)
+    if routine == "ai1" or routine == "all":
+        time += compute_advanced_indexing_1d(steps, N, timing)
+    if routine == "ai2" or routine == "all":
+        time += compute_advanced_indexing_2d(steps, N, timing)
+    if routine == "ai3" or routine == "all":
+        time += compute_advanced_indexing_3d(steps, N, timing)
+    if timing:
+        print("Total Elapsed Time: " + str(time) + " ms")
+    return time
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-n",
+        "--num",
+        type=int,
+        default=100,
+        dest="N",
+        help="number of elements in one dimension",
+    )
+    parser.add_argument(
+        "-i",
+        "--iterations",
+        type=int,
+        default=100,
+        dest="I",
+        help="number of iterations to run the algorithm for",
+    )
+    parser.add_argument(
+        "-t",
+        "--time",
+        dest="timing",
+        action="store_true",
+        help="perform timing",
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        dest="verbose",
+        action="store_true",
+        help="print verbose output",
+    )
+    parser.add_argument(
+        "-b",
+        "--benchmark",
+        type=int,
+        default=1,
+        dest="benchmark",
+        help="number of times to benchmark this application (default 1 - "
+        "normal execution)",
+    )
+    parser.add_argument(
+        "-r",
+        "--routine",
+        default="all",
+        dest="routine",
+        help="name of the index routine to test",
+    )
+    args, unknown = parser.parse_known_args()
+    print("Warning, unrecognized arguments: ", unknown)
+    run_benchmark(
+        run_indexing_routines,
+        args.benchmark,
+        "Core",
```

Comment:
I would call the benchmark suite "Indexing Routines"

Reply:
And maybe also note which benchmark it is, e.g. `"Indexing routines: " + args.routine`